### PR TITLE
MAGN-4745 Colour the nodes according to the delta state

### PR DESCRIPTION
**Pull request by ramramps**
The users have no idea of Node execution plan, when a value of a node is changed.

This fix address the node execution plan. When Show Node Execution is set on the Debug menu, 
users can see the nodes with Green boundary, meaning those nodes will be executed on the next run. 
Nodes will retain its original state after the run. 

- [ ] @lukechurch 
- [ ] @junmendoza 
- [x] @ikeough

### CODE REVIEW

**Comment by ramramps** [comment]
@junmendoza  - Please review the node execution path.

**Comment by ramramps** [comment]
@ikeough  - It took really long time to have these issues fixed. Getting the changes made by Jun and merging with new master (after the code re factoring from peter and stephen) took sometime. 
I hope I have addressed all the issues we discussed last time.

**Comment by ramramps** [comment]
@ikeough - Had discussion with Lili and Zach on the color. They came up with yellow, and we tried to see it through a large graph and it looked better than green. If you have any other alternative color, please let me know.

**Comment by ikeough** [comment]
@ramramps Can you fix the merge conflict so I can give it a try? Thanks!

**Comment by RodRecker** [comment]
@ikeough Will you be able to resolve this with Ram out for a few days?

**Comment by ikeough** [comment]
@RodRecker Yeah. I'll merge it in and take a look.

**Comment by ramramps** [comment]
@ikeough 
I believe I made all the changes what we discussed. Please take a look and let me know whether I have to make any other changes

**Comment by ikeough** [comment]
Thanks @ramramps! LGTM.

**Comment by ikeough** [comment]
@ramramps I need to revert this. It is causing an infinite loop on Workspace save. See the WorkspaceModel constructor where you added `WorkspaceSaved += OnWorkspaceSaved;`.

**Comment by ikeough** [comment]
@ramramps Execution preview is also still enable in Run Automatically. To test, start a new file, while running in debug mode. Execution preview will still be enabled and checked in the menu, but grayed out so you can't select it :)

**Comment by ke-yu** [comment]
https://github.com/DynamoDS/Dynamo/pull/3691 to remove this event handler.

File: src/Engine/ProtoScript/Runners/LiveRunner.cs [hunk 1]
```diff
@@ -373,6 +373,7 @@ private IEnumerable<AssociativeNode> GetDeltaAstListDeleted(IEnumerable<Subtree>
         private IEnumerable<AssociativeNode> GetDeltaAstListAdded(IEnumerable<Subtree> addedSubTrees)
         {
             var deltaAstList = new List<AssociativeNode>();
+            currentSubTreeList = new Dictionary<Guid, Subtree>();
```
Comment (ke-yu) [inline src/Engine/ProtoScript/Runners/LiveRunner.cs]: This causes some LiveRunner test cases fail. You may take to @junmendoza to confirm about this change...

**Pull request merged**